service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/AzureStorageAccessService.java

```diff
@@ -115,12 +103,15 @@ public AzureSasBundle createAzureStorageContainerSasToken(
     features.azureEnabledCheck();
 
     BlobContainerSasPermission blobContainerSasPermission =
-        getSasTokenPermissions(userRequest, storageContainerResource.getResourceId());
+        getSasTokenPermissions(
+            userRequest,
+            storageContainerResource.getResourceId(),
+            storageContainerResource.getCategory().getSamResourceName());
```

Comment:
This is the actual fix; we are threading in the resource's Sam resource name rather than relying on the previously hardcoded value.